app/models/job.ts

```diff
@@ -520,6 +520,23 @@ export class Job extends DBRecord implements JobRecord {
     }
   }
 
+  /**
+  * Returns the number of input granules for the given jobID
+  *
+  * @param tx - the database transaction to use for querying
+  * @param jobID - the jobID for the job that should be retrieved
+  * @param getLinks - if true include the job links when returning the job
+  * @param lock - if true lock the row in the jobs table
+  * @returns the Job with the given JobID or null if not found
+  */
+  static async getNumInputGranules(tx: Transaction, jobID: string): Promise<number> {
```

Comment:
Initially I did not have it in a transaction, but it's being called from two places one of which is in a transaction and trying to get a database connection when you are already in a transaction is a problem when running the tests with sqlite. So anyway I just made things straightforward and used a passed in transaction like we do just about 100% of the time. We'll need a bigger effort to rework the way we do transactions and I decided not to mess with things for this PR.

Reply:
You have me wondering about `numInputGranules` - is that getting updated if CMR returns fewer items than expected? As long as it is getting updated then everything should work fine. Even if it doesn't get updated... I think we'll be ok and will finish getting all the items - it would only be a problem if we allowed more granules than the initial query, which I remember we do not allow.